server/test/repository/ProgramRepositoryTest.java

```diff
@@ -100,6 +111,42 @@ public void lookupProgram_usesCacheWhenEnabled() {
     assertThat(programCache.get(String.valueOf(two.id))).hasValue(found);
   }
 
+  @Test
+  public void getFullProgramDefinitionFromCache_getsFromCacheWhenPresent() {
+    Mockito.when(mockSettingsManifest.getQuestionCacheEnabled()).thenReturn(true);
+    ProgramModel program = resourceCreator.insertActiveProgram("testInCache");
+    repo.setFullProgramDefinitionCache(program.id, program.getProgramDefinition());
+    Optional<ProgramDefinition> programDefFromCache =
```

Comment:
Nitpick (non-blocking): It's nice to leave an empty line between arrange / act / assert stages, so might be just a tad more readable if you have an empty line above the `Optional<ProgramDefinition> programDefFromCache = repo.getFullProgramDefinitionFromCache(program);` line.

Reply:
Good call - updated!